base_layer/wallet/tests/output_manager_service/service.rs

```diff
@@ -881,3 +881,45 @@ fn sending_transaction_with_short_term_clear_sqlite_db() {
 
     sending_transaction_with_short_term_clear(OutputManagerSqliteDatabase::new(connection));
 }
+
+fn create_coin_split<T: Clone + OutputManagerBackend + 'static>(backend: T) {
+    let factories = CryptoFactories::default();
+    let mut runtime = Runtime::new().unwrap();
+    let (mut oms, _, _, _) = setup_output_manager_service(&mut runtime, backend.clone());
+
+    let val1 = 6_000 * uT;
+    let val2 = 7_000 * uT;
+    let val3 = 8_000 * uT;
+    let (_ti, uo1) = make_input(&mut OsRng.clone(), val1, &factories.commitment);
+    let (_ti, uo2) = make_input(&mut OsRng.clone(), val2, &factories.commitment);
+    let (_ti, uo3) = make_input(&mut OsRng.clone(), val3, &factories.commitment);
+    assert!(runtime.block_on(oms.add_output(uo1)).is_ok());
+    assert!(runtime.block_on(oms.add_output(uo2)).is_ok());
+    assert!(runtime.block_on(oms.add_output(uo3)).is_ok());
+
+    let fee_per_gram = MicroTari::from(25);
+    let split_count = 8;
+    let (_tx_id, coin_split_tx, fee, amount) = runtime
+        .block_on(oms.create_coin_split(1000.into(), 8, fee_per_gram, None))
+        .unwrap();
+    assert_eq!(coin_split_tx.body.inputs().len(), 2);
+    assert_eq!(coin_split_tx.body.outputs().len(), split_count + 1);
+    assert_eq!(fee, Fee::calculate(fee_per_gram, 1, 2, split_count + 1));
+    assert_eq!(amount, val1 + val2);
+}
+
```

Comment:
You don't test the no-change case, which I think may have a bug as written

Reply:
Added a a no change test and fixed the bug.